src/torrent.cpp

```diff
@@ -11579,6 +11579,7 @@ namespace libtorrent
 	{
 		if (!m_apply_ip_filter) return;
 		if (!m_peer_list) return;
+		if (!m_ip_filter) return;
```

Comment:
The problem was with the line below `m_peer_list->apply_ip_filter(*m_ip_filter, &st, banned);`, it causes memory crashes